packages/editor-ui/src/components/Node.vue

```diff
@@ -328,6 +341,9 @@ export default mixins(
 		nodeRunData(newValue) {
 			this.$emit('run', {name: this.data.name, data: newValue, waiting: !!this.waiting});
 		},
+		workflowDataItems(dataItemsCount: number) {
+			this.checkPinDataDiscoveryFlow(dataItemsCount);
+		},
```

Comment:
Open question: Wondering if this really needs to be a watcher? Can we latch onto anything else in the canvas that does not require querying local storage every time `workflowDataItems` changes? Concerned that we are watching a frequently changing reactive property for something that only ever needs to happen once, unless I am misunderstanding.

Reply:
Good point. Changed flow with `watch/unwatch`.